### PR TITLE
setup project

### PR DESCRIPTION
**Pull request by hagan199**
- Use callbacks and promises.
- Learn how to use proper ES6 syntax.
- Use ES6 modules to write modular JavaScript.
- Use webpack to bundle JavaScript.

### CODE REVIEW

**Review by Ismailco — COMMENTED** [review]
Hi @hagan199, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that should be built using Gitflow, kindly make sure to use the gitflow instead of GitHub flow, your changes should be present in a feature branch and the PR should merge the changes from the feature branch to the `development` branch

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Review by V-Blaze — CHANGES_REQUESTED** [review]
Hi @hagan199  , 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

dcentl8-9b96e129-9725-401b-ac34-2feb990202a0

## Invalid Code Review Request

## You have submitted a project, that doesn't follow Gitflow. The previous reviewer pointed this out, I'll suggest you follow the requirement and implement it before requesting another review. To Guide you further, this is an article from [Microvers on GItflow](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/git-github/articles/gitflow.md), you can go through the article for a better understanding.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Review by Davidosky007 — COMMENTED** [review]
Hi @hagan199 👋, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

## You have submitted a project, that doesn't follow proper Gitflow partern , Kindly create a pull request to compare your feature branch with the development branch and submit the link to that pull request via your Dashboard, your PR should look like this bellow 👇 ( Dev beign your default branch and main-file your feature branch)
image


Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Review by TedLivist — COMMENTED** [review]
## STATUS: INVALID CODE REVIEW REQUEST 🛑 🛑

Hi @hagan199,

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

### Kindly request a merge to the dev branch, not the main branch, in keeping with the required gitflow.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: README.md [hunk 1]
```diff
@@ -1,2 +1,136 @@
...
+- clone the project using by running the following command :
+
+`git clone`
+`https://github.com/hagan199/Leaderboard.git`
+
+###  Install
+
+ ```install
+ npm install
+ ```
```
Comment (topeogunleye) [inline README.md:72]: - [ ] I kindly suggest that you add the step for usage of your web app. for example
[OPTIONAL]
```
### Usage

To run the project, execute the following command:


- Start the development server
`   ```sh
`   npm start
`   ```
`
```

**Review by topeogunleye — APPROVED** [review]
Hi @hagan199, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

dwayne-the-rock

## (Highlights) Good Points: 💪 
* No linter errors. :heavy_check_mark:
* Followed Github flow :heavy_check_mark:
* PR has a good title  :heavy_check_mark:
* Great UI design implementation ✔️ 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

You can also consider:
_Please kindly check my review comments_

Your branch name dev does not follow the convention of being named after the feature or action that you implemented. For example, you could name your branch `feature/setup_projoct`. This makes it easier to track the changes and the purpose of each branch.

- [ ]  Next time, and for future commits, I kindly suggest that you complete your commit messages.
:x: Bad Commit Messages :x:

image

### ✔️ Good Commit Messages: ✔️
✔️  Add fix for dark mode toggle state
✔️  Update README file.
✔️  Add styles to h1.

- [ ] Please refer to this article: please read it and apply it in your work from now on: [Professional repo - basic rules](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/documentation/articles/professional_repo_rules.md);



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag @topeogunleye in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: README.md (hunk 1 above)
Comment (hagan199) [reply README.md:72]: Noted

**Pull request merged**